VOKMockUrlProtocol.m

```diff
@@ -271,42 +296,49 @@ - (VOKMockUrlProtocolResponseAndDataContainer *)responseAndDataFromHTTPResponseS
+            DLOG(@"Serving mock data from complete HTTP response file (%@)", filePath);
+            return [self responseAndDataFromHTTPResponseString:fileContents];
+        }
     }
     
     // Otherwise, look for a JSON data file.
-    filePath = [[NSBundle bundleForClass:[self class]] pathForResource:resourceName
-                                                                ofType:@"json"
-                                                           inDirectory:MockDataDirectory];
-    NSData *data = [NSData dataWithContentsOfFile:filePath];
-    if (data) {
-        // We've got a JSON data file, so send it.
-        DLOG(@"Serving mock data from JSON response body file (%@)", filePath);
-        NSHTTPURLResponse *response = [[NSHTTPURLResponse alloc] initWithURL:self.request.URL
-                                                                  statusCode:kHTTPStatusCodeOK
-                                                                 HTTPVersion:@"HTTP/1.1"
-                                                                headerFields:@{
-                                                                               @"Content-type": @"text/json",
-                                                                               }];
-        return [VOKMockUrlProtocolResponseAndDataContainer containerWithResponse:response
-                                                                            data:data];
+    for (NSString *resourceName in resourceNames) {
+        filePath = [[NSBundle bundleForClass:[self class]] pathForResource:resourceName
+                                                                    ofType:@"json"
+                                                               inDirectory:MockDataDirectory];
+        NSData *data = [NSData dataWithContentsOfFile:filePath];
+        if (data) {
+            // We've got a JSON data file, so send it.
+            DLOG(@"Serving mock data from JSON response body file (%@)", filePath);
+            NSHTTPURLResponse *response = [[NSHTTPURLResponse alloc]
+                                           initWithURL:self.request.URL
+                                           statusCode:kHTTPStatusCodeOK
+                                           HTTPVersion:@"HTTP/1.1"
+                                           headerFields:@{
+                                                          HTTPHeaderContentType: HTTPHeaderContentTypeJson,
+                                                          }];
+            return [VOKMockUrlProtocolResponseAndDataContainer containerWithResponse:response
+                                                                                data:data];
+        }
     }
     
     // Otherwise, failure.
-    DLOG(@"failed to get mock data for resource name: \"%@\"", resourceName);
+    for (NSString *resourceName in resourceNames) {
+        DLOG(@"failed to get mock data for resource name: \"%@\"", resourceName);
```

Comment:
Actually, no conflict with `DLog`, since macros are case-sensitive, but there's a conditional definition of `DLOG` so that it's defined only when it hasn't already been defined.